### PR TITLE
Dirty fix mkchain for release

### LINKED ISSUE #66: "mkchain my-chain" in the readme fails - needs updating
The readme quickstart section of the readme contains the following:
```
mkchain my-chain
```
which when executed as instructed fails with the following message:
```
mkchain my-chain
usage: mkchain [-h] [--version] {generate-constants,create,invite} ...
mkchain: error: argument action: invalid choice: 'mkchain' (choose from 'generate-constants', 'create', 'invite')
```
The readme needs updating

### LINKED ISSUE #71: mkchain fails to run minikube's docker
https://github.com/tqtezos/tezos-k8s/blob/0b2ed250ae44ec23f492533130081dc511c1ecb5/mkchain/tqchain/mkchain.py#L31

executing the following
```
mkchain $CHAIN_NAME --zerotier-network $ZT_NET --zerotier-token $ZT_TOKEN
```
results in the error
```
bozon@buba test % minikube start       
😄  minikube v1.16.0 on Darwin 11.1
✨  Using the hyperkit driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🏃  Updating the running hyperkit "minikube" VM ...
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
bozon@buba test % eval $(minikube docker-env)
bozon@buba test % mkdir mkchain && cd mkchain
python3 -m venv .venv
source .venv/bin/activate

(.venv) bozon@buba mkchain % 
(.venv) bozon@buba mkchain % pip install mkchain==0.0.1
Collecting mkchain==0.0.1
  Using cached mkchain-0.0.1-py3-none-any.whl (5.0 kB)
Collecting pyyaml
  Using cached PyYAML-5.3.1.tar.gz (269 kB)
Using legacy 'setup.py install' for pyyaml, since package 'wheel' is not installed.
Installing collected packages: pyyaml, mkchain
    Running setup.py install for pyyaml ... done
Successfully installed mkchain-0.0.1 pyyaml-5.3.1
WARNING: You are using pip version 20.2.3; however, version 20.3.3 is available.
You should consider upgrading via the '/Users/bozon/Documents/TQ/test/mkchain/.venv/bin/python3 -m pip install --upgrade pip' command.
(.venv) bozon@buba mkchain % export PYTHONUNBUFFERED=x
(.venv) bozon@buba mkchain % export CHAIN_NAME=avb     
(.venv) bozon@buba mkchain % mkchain $CHAIN_NAME --zerotier-network $ZT_NET --zerotier-token $ZT_TOKEN
Traceback (most recent call last):
  File "/Users/bozon/Documents/TQ/test/mkchain/.venv/bin/mkchain", line 8, in <module>
    sys.exit(main())
  File "/Users/bozon/Documents/TQ/test/mkchain/.venv/lib/python3.9/site-packages/tqchain/mkchain.py", line 125, in main
    "genesis_chain_id": get_genesis_vanity_chain_id(),
  File "/Users/bozon/Documents/TQ/test/mkchain/.venv/lib/python3.9/site-packages/tqchain/mkchain.py", line 48, in get_genesis_vanity_chain_id
    run_docker(
  File "/Users/bozon/Documents/TQ/test/mkchain/.venv/lib/python3.9/site-packages/tqchain/mkchain.py", line 19, in run_docker
    return subprocess.check_output(
  File "/usr/local/Cellar/python@3.9/3.9.1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/local/Cellar/python@3.9/3.9.1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'docker run --entrypoint flextesa --rm registry.gitlab.com/tezos/flextesa:01e3f596-run vani "" --seed 4X4M58LLL043JZWS --first --machine-readable csv' returned non-zero exit status 127.
```

### PR DESCRIPTION
**Pull request by harryttd**
Ideally we want to remove docker completely from mkchain

closes #71

### CODE REVIEW

**Comment by nicolasochem** [comment]
Now each time I run `mkchain`, it requires me to be authenticated to docker hub and prints a lot of lines of output. I think it's better to check whether the image is already pulled, and pull if it isn't.

File: mkchain/tqchain/mkchain.py [hunk 1]
```diff
@@ -105,10 +105,16 @@ def get_args():
...
 
 def pull_docker_images(images):
-    print("Pulling docker images...")
     for image in images:
-        subprocess.check_output(f"docker pull {image}", shell=True)
-    print("Done pulling docker images")
+        has_image_return_code = subprocess.run(
+            f"docker inspect --type=image {image} > /dev/null 2>&1", shell=True
+        ).returncode
+        if has_image_return_code == 1:
```
Comment (nicolasochem) [inline mkchain/tqchain/mkchain.py:112]: Please make it `!=0`, my podman returns 125 error code when image is not found.

**Review by nicolasochem — CHANGES_REQUESTED** [review]

File: mkchain/tqchain/mkchain.py (hunk 1 above)
Comment (harryttd) [reply mkchain/tqchain/mkchain.py:112]: good point

**Review by nicolasochem — APPROVED** [review]
Works as expected now



```
[venv] nochem@peck ~/workspace/tezos-k8s () $ mkchain nochem --number-of-nodes 5 --number-of-bakers 3
Pulling docker image registry.gitlab.com/tezos/flextesa:01e3f596-run
Done pulling docker image registry.gitlab.com/tezos/flextesa:01e3f596-run
Wrote chain creation constants to /home/nochem/workspace/tezos-k8s/nochem_values.yaml
Wrote chain invitation constants to /home/nochem/workspace/tezos-k8s/nochem_invite_values.yaml
[venv] nochem@peck ~/workspace/tezos-k8s () $ mkchain nochem --number-of-nodes 5 --number-of-bakers 3
Wrote chain creation constants to /home/nochem/workspace/tezos-k8s/nochem_values.yaml
Wrote chain invitation constants to /home/nochem/workspace/tezos-k8s/nochem_invite_values.yaml
```

**Pull request merged**